coordinator/internal/logic/auth/login.go

```diff
@@ -1,25 +1,111 @@
 package auth
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/gin-gonic/gin"
+	"github.com/scroll-tech/go-ethereum/log"
 	"gorm.io/gorm"
 
+	"scroll-tech/common/types/message"
+	"scroll-tech/common/version"
+
+	"scroll-tech/coordinator/internal/config"
+	"scroll-tech/coordinator/internal/logic/verifier"
 	"scroll-tech/coordinator/internal/orm"
+	"scroll-tech/coordinator/internal/types"
+)
+
+var (
+	proofChunkTypeString  = fmt.Sprintf("%d", message.ProofTypeChunk)
+	proofBatchTypeString  = fmt.Sprintf("%d", message.ProofTypeBatch)
+	proofBundleTypeString = fmt.Sprintf("%d", message.ProofTypeBundle)
 )
 
 // LoginLogic the auth logic
 type LoginLogic struct {
+	cfg          *config.Config
 	challengeOrm *orm.Challenge
+	chunkVks     map[string]struct{}
+	batchVKs     map[string]struct{}
+	bundleVks    map[string]struct{}
 }
 
 // NewLoginLogic new a LoginLogic
-func NewLoginLogic(db *gorm.DB) *LoginLogic {
-	return &LoginLogic{
+func NewLoginLogic(db *gorm.DB, cfg *config.Config, vf *verifier.Verifier) *LoginLogic {
+	l := &LoginLogic{
+		cfg:          cfg,
+		chunkVks:     make(map[string]struct{}),
+		batchVKs:     make(map[string]struct{}),
+		bundleVks:    make(map[string]struct{}),
 		challengeOrm: orm.NewChallenge(db),
 	}
+
+	for _, vk := range vf.ChunkVKMap {
+		l.chunkVks[vk] = struct{}{}
+	}
+
+	for _, vk := range vf.BatchVKMap {
+		l.batchVKs[vk] = struct{}{}
+	}
+
+	for _, vk := range vf.BundleVkMap {
+		l.bundleVks[vk] = struct{}{}
```

Comment:
It seems that the vks have not been prepared when initializing this struct.

Reply:
simplified in [95448cf](https://github.com/scroll-tech/scroll/pull/1399/commits/95448cf25f22bd94a6fdf523a724120dbb1ea463).

Reply:
seems that you replied to a wrong thread. @colinlyguo 
I have resolved the one related to your commit

Reply:
got it.